samples/embedded-map/src/components/EmbeddedMap/EmbeddedMapModel.ts

```diff
@@ -6,49 +6,69 @@ import {
 } from "@vertigis/web/models";
 import { throttle } from "@vertigis/web/ui";
 import Point from "esri/geometry/Point";
+import { whenDefinedOnce } from "esri/core/watchUtils";
 import { Viewer } from "mapillary-js";
 
 /**
  *  Convert Mapillary bearing to a Scene's camera rotation.
  *  @param bearing Mapillary bearing in degrees (degrees relative to due north).
  *  @returns Scene camera rotation in degrees (degrees rotation of due north).
  */
-
 function getCameraRotationFromBearing(bearing: number): number {
     return 360 - bearing;
 }
 
+interface MapillaryCamera {
+    latitude: number;
+    longitude: number;
+    heading: number;
+    tilt: number;
+    fov: number;
+}
+
+interface GcxMap extends MapExtension {
+    view: __esri.SceneView;
+}
+
 @serializable
 export default class EmbeddedMapModel extends ComponentModelBase {
-    private _map: MapExtension | undefined;
-    get map(): MapExtension | undefined {
-        return this._map;
-    }
-    @importModel("map-extension")
-    set map(instance: MapExtension | undefined) {
-        if (instance === this._map) {
-            return;
-        }
+    // For demonstration purposes only.
+    // Replace this with your own client ID from mapillary.com
+    readonly mapillaryKey =
+        "ZU5PcllvUTJIX24wOW9LSkR4dlE5UTo3NTZiMzY4ZjBlM2U2Nzlm";
 
-        // If an instance already exists, clean it up first.
-        if (this._map) {
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            this._unsyncMaps();
-        }
+    readonly imageQueryUrl = "https://a.mapillary.com/v3/images";
+    readonly searchRadius = 500; // meters
 
-        this._map = instance;
+    private _currentPosition: { latitude: number; longitude: number };
+    private _lastMarkerUpdate: any;
```

Comment:
Looks like we're expecting this to be an object with a `geometry` property, and that appears to be the only property we care about. Would you mind changing to the type to `Geometry` or even a `{ latitude: number; longitude: number }` interface as above, instead of `any`? It can be destructured at the time it's set in the event handler.

Reply:
Looks good.